### PR TITLE
pyegan

### LINKED ISSUE #312: connect egan to OpenAI baselines implementations
1- add reset to egan
2- PyEgan
3- EganGym(Egan)
   - deal with step blocking and passing data

### PR DESCRIPTION
**Pull request by etuleu**
Fixes #312 

TODO:

1. bazel build
2. test files for PyEgan and EganEnv

### CODE REVIEW

File: worlds/pyegan/EganEnv.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    def close(self):
+        logger.info(f"close called")
+
+
+def main():
+    egan = EganEnv()
+    # model = A2C('CnnPolicy', egan).learn(total_timesteps=1000)
+    model = PPO("MlpPolicy", egan, verbose=1).learn(total_timesteps=1000)
+
```
Comment (wumpusman) [inline worlds/pyegan/EganEnv.py:55]: Make a separate main for the model

File: worlds/pyegan/PyEgan.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        # this blocks waiting on egan to call step
+        self.server.serve()
+
+
+def main():
+    pyegan = PyEgan()
+
+
+if __name__ == "__main__":
+    main()
```
Comment (wumpusman) [inline worlds/pyegan/PyEgan.py:55]: remove main, since this is for testing.

File: worlds/pyegan/EganEnv.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+class EganEnv(gym.Env):
+    """Egan Environment that follows gym interface"""
+
+    metadata = {"render.modes": ["human"]}
+
+    def __init__(self):
+        super().__init__()
+        # Define action and observation space
+        # They must be gym.spaces objects
+        N_DISCRETE_ACTIONS = 2
```
Comment (etuleu) [inline worlds/pyegan/EganEnv.py:22]: there are more than 2 actions in egan. update this

File: worlds/pyegan/EganEnv.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+from PyEgan import PyEgan
+
+logger = logger.get_logger(__name__)
+
+
+class EganEnv(gym.Env):
+    """Egan Environment that follows gym interface"""
+
+    metadata = {"render.modes": ["human"]}
```
Comment (etuleu) [inline worlds/pyegan/EganEnv.py:16]: update this also

**Review by wumpusman — APPROVED** [review]

**Pull request merged**